From 444089eec6042250ce3a7184cb09bd8a2ab16808 Mon Sep 17 00:00:00 2001
From: Vladimir Serbinenko <phcoder@gmail.com>
Date: Thu, 6 Jul 2023 17:13:50 +0200
Subject: [PATCH 30/57] ZFS: Don't iterate over null objsets

Reading them is harmless but useless as they are empty by definition

Signed-off-by: Vladimir Serbinenko <phcoder@gmail.com>
Reviewed-by: Daniel Kiper <daniel.kiper@oracle.com>
---
 grub-core/fs/zfs/zfs.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/grub-core/fs/zfs/zfs.c b/grub-core/fs/zfs/zfs.c
index bd49ddb9e..221c914cb 100644
--- a/grub-core/fs/zfs/zfs.c
+++ b/grub-core/fs/zfs/zfs.c
@@ -4157,6 +4157,9 @@ iterate_zap_fs (const char *name, grub_uint64_t val,
   grub_err_t err;
   struct grub_dirhook_info info;
 
+  if (name[0] == 0 && val == 0)
+    return 0;
+
   dnode_end_t mdn;
   err = dnode_get (&(ctx->data->mos), val, 0, &mdn, ctx->data);
   if (err)
-- 
2.42.0

